### PR TITLE
[725] Port over course salary editing functionality

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/u4mkCREO/725-migrate-courses-course-show-page-edit-course-length-salary

### Changes proposed in this pull request

- Handle editing course length information when course is salary based as per this PR https://github.com/DFE-Digital/teacher-training-api/pull/2524/files
- DRY up shared behaviour across these two types of forms via controllers/form objects
- Increase test coverage to deal with both types in feature spec

### Guidance to review

Original implementation here: https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2CK/2022/courses/L835

migrated version: https://teacher-training-api-pr-2535.london.cloudapps.digital/publish/organisations/2CK/2022/courses/L835

Check:

You can update course length and salary

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Migrate Courses - Course show page - edit course length & salary](https://trello.com/c/u4mkCREO/725-migrate-courses-course-show-page-edit-course-length-salary)

**Review by goodviber — APPROVED** [review]
LGTM, maybe have Sau have a look incase I am missing some context?

File: app/views/publish/courses/_course_length_field.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      checked: form_object.other_course_length?,
+      label: { text: "Other" },
+      data: { qa: "course_course_length_other" }) do %>
+
+      <%= f.govuk_text_field(:course_length_other_length,
+        label: { text: "Course length", size: "s" },
+        width: 20,
+        data: { qa: "course_course_length_other_length" }) %>
+    <% end %>
+<% end %>
```
Comment (defong) [inline app/views/publish/courses/_course_length_field.html.erb:25]: newline please

File: app/views/publish/courses/salary/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<% page_title = "Course length and fees" %>
```
Comment (defong) [inline app/views/publish/courses/salary/edit.html.erb:1]:
```suggestion
<% page_title = "Course length and salary" %>
```

image

**Review by defong — CHANGES_REQUESTED** [review]

**Review by defong — APPROVED** [review]

**Pull request merged**